cdc_stream/main.cc

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <vector>
+
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/flags/usage.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_split.h"
+#include "cdc_stream/local_assets_stream_manager_client.h"
+#include "common/log.h"
+#include "common/path.h"
+#include "grpcpp/channel.h"
+#include "grpcpp/security/credentials.h"
+
+ABSL_FLAG(uint16_t, service_port, 44432,
+          "Port to use while connecting to the local asset stream service.");
+ABSL_FLAG(
+    uint16_t, ssh_port, 22,
+    "Port to use while connecting to the remote instance being streamed to.");
+ABSL_FLAG(std::string, ssh_command, "",
+          "Path and arguments of ssh command to use, e.g. "
+          "C:\\path\\to\\ssh.exe -F config_file. Can also be specified by the "
+          "CDC_SSH_COMMAND environment variable.");
+ABSL_FLAG(std::string, scp_command, "",
+          "Path and arguments of scp command to use, e.g. "
+          "C:\\path\\to\\scp.exe -F config_file. Can also be specified by the "
+          "CDC_SCP_COMMAND environment variable.");
+ABSL_FLAG(int, verbosity, 2, "Verbosity of the log output");
+
+namespace {
+
+constexpr char kHelpText[] = R"!(Stream files from a Windows to a Linux device
+
+Usage: cdc_stream [flags] start windows_dir [user@]host:linux_dir
+       Streams the Windows directory windows_dir to directory linux_dir on the
+       Linux host using SSH username user.
+ 
+       cdc_stream [flags] stop host
```

Comment:
We can stream multiple sessions to the same host, as long as the target directories are different. This actually uses one cache for both sessions, so the cache cleanup won't work properly since each instance keeps a separate track of the files added. This needs to be revisited at some point, but it's not a big deal for now.